jetbrains: 2025.1.1 -> 2025.1.5

ners opened the pull request:
## Things done

Fixes https://github.com/NixOS/nixpkgs/issues/425328

```
jetbrains.clion: 2025.1.2 -> 2025.1.4
jetbrains.dataspell: 2025.1.1 -> 2025.1.2.1
jetbrains.goland: 2025.1.2 -> 2025.1.4
jetbrains.idea-community: 2025.1.2 -> 2025.1.4.1
jetbrains.idea-ultimate: 2025.1.2 -> 2025.1.4.1
jetbrains.phpstorm: 2025.1.2 -> 2025.1.4.1
jetbrains.pycharm-community: 2025.1.2 -> 2025.1.3.1
jetbrains.pycharm-professional: 2025.1.2 -> 2025.1.3.1
jetbrains.rider: 2025.1.2 -> 2025.1.4
jetbrains.ruby-mine: 2025.1.2 -> 2025.1.4.1
jetbrains.rust-rover: 2025.1.3 -> 2025.1.5
jetbrains.webstorm: 2025.1.2 -> 2025.1.4.1
```

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MattSturgeon commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 419026`

Logs: https://github.com/MattSturgeon/nixpkgs-review-gha/actions/runs/15879156940


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.rider</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-bin)</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>jetbrains.clion</summary>
<pre>searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_codecs_iso2022.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/unicodedata.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_codecs_cn.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_datetime.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_codecs_hk.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/array.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_lsprof.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_multibytecodec.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_testimportmultiple.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/binascii.cpython-38-aarch64-linux-gnu.so
    libz.so.1 -> found: /nix/store/4naij40hybv9hdsfvvf2b8h437w67bb1-zlib-1.3.1/lib
setting RPATH to: /nix/store/4naij40hybv9hdsfvvf2b8h437w67bb1-zlib-1.3.1/lib
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_pickle.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_elementtree.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/resource.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/gdb/linux/aarch64/lib/python3.8/lib-dynload/_queue.cpython-38-aarch64-linux-gnu.so
auto-patchelf: 1 dependencies could not be satisfied
error: auto-patchelf could not satisfy dependency libxml2.so.2 wanted by /nix/store/hih0zz503dmf94sv01gvm1l08jnfb85p-clion-2025.1.3/clion/bin/lldb/linux/aarch64/lib/liblldb.so
auto-patchelf failed to find all the required dependencies.
Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.</pre>
</details>
<details>
<summary>jetbrains.rider</summary>
<pre>searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_datetime.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_codecs_hk.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/array.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_lsprof.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_multibytecodec.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_testimportmultiple.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/binascii.cpython-38-aarch64-linux-gnu.so
    libz.so.1 -> found: /nix/store/4naij40hybv9hdsfvvf2b8h437w67bb1-zlib-1.3.1/lib
setting RPATH to: /nix/store/4naij40hybv9hdsfvvf2b8h437w67bb1-zlib-1.3.1/lib
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_pickle.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_elementtree.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/resource.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/lib-dynload/_queue.cpython-38-aarch64-linux-gnu.so
searching for dependencies of /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/python3.8/site-packages/lldb/_lldb.so
    liblldb.so -> found: /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib
setting RPATH to: /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib
auto-patchelf: 1 dependencies could not be satisfied
error: auto-patchelf could not satisfy dependency libxml2.so.2 wanted by /nix/store/i6bbkj0zr88nm3hn84why300k6hjvrx4-rider-2025.1.3/rider/plugins/cidr-debugger-plugin/bin/lldb/linux/aarch64/lib/liblldb.so
auto-patchelf failed to find all the required dependencies.
Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.</pre>
</details>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-bin)</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-bin)</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>jetbrains.rider</summary>
<pre>Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/brr8y5j32k2whyb82brk99x8v2y9xz8c-JetBrains.Rider-2025.1.3-aarch64.dmg
source root is .
setting SOURCE_DATE_EPOCH to timestamp 1750862241 of file "./Rider.app/Contents/plugins/cidr-debugger-plugin/bin/lldb/mac/x64/LLDB.framework/Headers/SBWatchpointOptions.h"
warning: file "./Rider.app/Contents/plugins/cidr-debugger-plugin/bin/lldb/mac/x64/LLDB.framework/Headers/SBWatchpointOptions.h" may be generated; SOURCE_DATE_EPOCH may be non-deterministic
unpackPhase completed in 1 minutes 23 seconds
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
no Makefile or custom buildPhase, doing nothing
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
cp: error writing '/nix/store/iyappsxq6a8j31c0z65rcljcc4f041s6-rider-2025.1.3/Applications/Rider.app/Contents/plugins/fullLine/full-line-model-ws.zip': No space left on device</pre>
</details>
<details>
<summary>jetbrains.ruby-mine</summary>
<pre>Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/n1kphxls6alx1463321c4jxiv2x12lpr-RubyMine-2025.1.3-aarch64.dmg
source root is .
unpackPhase completed in 43 seconds
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
no Makefile or custom buildPhase, doing nothing
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
cp: error writing '/nix/store/7dd9j1hmnjqams3f8fhgzd72964rqx00-ruby-mine-2025.1.3/Applications/RubyMine.app/Contents/plugins/ruby/rubysigs/rubysigs32/stdlib/tempfile/0/tempfile.rbs': No space left on device
cp: cannot create directory '/nix/store/7dd9j1hmnjqams3f8fhgzd72964rqx00-ruby-mine-2025.1.3/Applications/RubyMine.app/Contents/plugins/ruby/rubysigs/rubysigs32/stdlib/time/0': No space left on device</pre>
</details>
</details>

ners commented:
@sdht0 thanks for the tip. I was able to easily add another `patchelf` case to Rider, Clion, and RustRover. I've also simplified the library finding.

theCapypara reviewed (changes requested):
Unfortunately CLion doesn't build anymore with these changes, see above.
I can only easily test x86_64-linux

MattSturgeon commented on pkgs/applications/editors/jetbrains/default.nix:
```diff
@@ -204,18 +204,11 @@ rec {
-                --replace-needed libcrypto.so.10 libcrypto.so
-
-              ls -d $PWD/bin/lldb/linux/*/lib/python3.8/lib-dynload/* |
+              find $out -name '*.so' |
```
> ```
> patchelf: missing ELF header
> ```

I'm guessing there are some libs that shouldn't be patched, so we need to filter them out somehow?

I wonder if the previous globs for `gdb` and `lldb` are more accurate than finding _all_ `.so` files in `$out`?

ners committed:
jetbrains: patch libxml2 dependency

MattSturgeon reviewed (approved):
Tested this with my system, which [uses](https://github.com/MattSturgeon/nix-config/blob/4c0c5447ce5fd2b49e6d2fe0d0a0893ecf6a46d7/modules/home-manager/intellij.nix#L23) `addPlugins` on `idea-community-bin`.

```sh
sudo nixos-rebuild switch \
  --flake 'github:MattSturgeon/nix-config' \
  --override-input nixpkgs 'github:NixOS/nixpkgs?ref=pull/419026/merge'
```

Intellij seems to launch and run fine.

---

I also ran nixpkgs-review:

## `nixpkgs-review` result for [#419026](https://github.com/NixOS/nixpkgs/pull/419026)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 419026`

Logs: https://github.com/MattSturgeon/nixpkgs-review-gha/actions/runs/16032474545


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-bin)</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-bin)</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-bin)</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

MattSturgeon commented on pkgs/applications/editors/jetbrains/default.nix:
```diff
@@ -347,20 +341,12 @@ rec {
-            )
+            ${patchSharedLibs}
+
+            for dir in $out/rider/ReSharperHost/linux-*; do
```
This path was `$out/rider/lib/ReSharperHost/linux-*` before. Is the change intentional?

Could be related to https://github.com/NixOS/nixpkgs/pull/419026#issuecomment-3034879811

```suggestion
            for dir in $out/rider/lib/ReSharperHost/linux-*; do
```

ners replied:
Thanks, that was indeed not intended.

tanordheim replied:
That solved it, just tested the most recent commits!

ners commented:
Thank you for all the feedback @MattSturgeon @tanordheim! :slightly_smiling_face:

theCapypara commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 419026`
Commit: `34dd640fe80974409739c571a923f9a9a6901265`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

theCapypara reviewed (approved):
LGTM on x86_64 Linux.
I tested launching Rider, PhpStorm and RustRover. For the latter two I also did basic functionality checks.

Thank you!

theCapypara commented:
Again, please do not merge this until somebody has confirmed the other three targets as working.

MattSturgeon commented:
> Again, please do not merge this until somebody has confirmed the other three targets as working.

In https://github.com/NixOS/nixpkgs/pull/419026#discussion_r2185482999 @tanordheim confirmed it working on aarch64 (I assume linux).

---

I'll run another nixpkgs-review to test the build on all systems. I can't test the other systems at runtime though.

JBassett commented:
> > Again, please do not merge this until somebody has confirmed the other three targets as working.
> 
> In [#419026 (comment)](https://github.com/NixOS/nixpkgs/pull/419026#discussion_r2185482999) @tanordheim confirmed it working on aarch64 (I assume linux).
> 
> I'll run another nixpkgs-review to test the build on all systems. I can't test the other systems at runtime though.

So can we get this merged?

theCapypara commented:
If somebody can at least confirm the IDEs work under aarch64-darwin, yes, IMO

john-rodewald commented:
I can’t run this on `aarch64-darwin`. It seems like we’re not building an executable.

The same thing happens on `nixos-unstable`, so I’m not sure if that’s a regression.

```
NIXPKGS_ALLOW_UNFREE=1 nix run --impure github:ners/nixpkgs/nixos-unstable#jetbrains.rider
error: unable to execute '/nix/store/2vq0vsjbw65qpk0797h5pylcf5z02j94-rider-2025.1.4/bin/rider': Exec format error

lipo -archs /nix/store/2vq0vsjbw65qpk0797h5pylcf5z02j94-rider-2025.1.4/bin/rider
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't figure out the architecture type of: /nix/store/2vq0vsjbw65qpk0797h5pylcf5z02j94-rider-2025.1.4/bin/rider

file /nix/store/2vq0vsjbw65qpk0797h5pylcf5z02j94-rider-2025.1.4/bin/rider
/nix/store/2vq0vsjbw65qpk0797h5pylcf5z02j94-rider-2025.1.4/bin/rider: ASCII text
```

MattSturgeon commented:
Is this related to the [Rosetta translation](https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment) feature for x86 support on arm macs... Looks like the wrapper script is a shell script, which is maybe tripping up Rosetta? 🤔

I agree this probably isn't a regression caused by this PR. I suspect switching to the binary variant of `makeWrapper` may solve the issue, though.

The nixpkgs manuals already gives _other_ examples of where using a binary wrapper is required on darwin; _those_ examples aren't relevant here, but this seems to be a similar issue in practice.

https://nixos.org/manual/nixpkgs/unstable/#fun-makeWrapper

john-rodewald commented:
Ah yeah, that sounds exactly right.

`sh /nix/store/2vq0vsjbw65qpk0797h5pylcf5z02j94-rider-2025.1.4/bin/rider` works like a charm. 😅

ners committed:
jetbrains: fix Darwin wrapper script

ners commented:
Thanks for the investigation @MattSturgeon. I've looked at some other Darwin wrappers (e.g. iterm2, libreoffice) and they always have a shebang in the wrapper. Tested building and running it on an aarch64-darwin, it works! :tada:

john-rodewald commented:
Can confirm that Rider launches on `aarch64-darwin` now. :)

ners commented:
With yesterday's rebase to master, I am no longer able to build any of the IDEs because `jetbrains-jdk-jcef-21.0.6-b895.109` fails to build with the following error:
```
./jb/project/tools/linux/scripts/mkimages_x64.sh: line 35: BOOT_JDK: unbound variable
```

I think this should be fixed before we merge.

ners commented:
I've fixed the `jetbrains-jdk` build and updated everything again.

aoli-al reviewed (approved):
jetbrains.jdk fix looks good to me.

jys1670 commented:
> I've fixed the `jetbrains-jdk` build and updated everything again.

Have you tried running any IDE with it?
There is PR with a similar fix, however right after starting, pycharm fails with internal error for me.
I don't have capacity to rebuild jdk and test right now.

jys1670 commented:
I can confirm PyCharm builds and starts with JDK changes from this PR. However, whenever I open a markdown file, a JCEF segfault occurs.
This does not happen with
```
separateDebugInfo = false;
__structuredAttrs = false;
```
override. However, it might be one of my patches that caused the issue.
@aoli-al Can you check that markdown works for you?

Side notes:
- Aqua and Writerside are both dead ides and can be removed
- fsnotifier should probably be used directly from nixpkgs (not via some roundabout way, as it has an essential patch for hardlinks)

MattSturgeon commented:
I've re-ran [this review](https://github.com/NixOS/nixpkgs/pull/419026#issuecomment-3075612436) ([here](https://github.com/MattSturgeon/nixpkgs-review-gha/actions/runs/16327444156)), and it is _still_ failing to build idea-community-src, pycharm-community-src, and animeko.

I see the same issue when [reviewing](https://github.com/MattSturgeon/nixpkgs-review-gha/actions/runs/16327438719) https://github.com/NixOS/nixpkgs/pull/425529, and the issue started when changing jetbrains-jdk, so that seems to be the issue.

IIUC, jetbrains-jdk is tightly coupled to the source-built IDEs, so it is usually updated in sync with them.

I'm not surprised that toggling __structuredAttrs would have unexpected consequences, since that is a wide-reaching change that impacts how derivation attrs are exposed as shell variables, meaning the various hooks, phases, etc must use different bash syntax (e.g. space-separated lists vs actual bash arrays).

aoli-al commented:
The `jetbrains.idea-community-src` build failure should be addressed here #425814. Feel free to also include the patch in this PR.

theCapypara commented:
I'm a bit confused on what is the actual state of the PR now. Am I understanding correctly that this PR currently blocks this?:

- https://github.com/NixOS/nixpkgs/pull/425529

Or does this PR contain other alternative changes and we need to settle on which ones to use? As I understand it the changes in this PR cause segfaults with markdown files? And I take it the changes in the above PR don't cause one?

And this PR fixes the src-build but is otherwise not relevant for this PR?:

- https://github.com/NixOS/nixpkgs/pull/425814

aoli-al commented:
Sorry for the confusion. #425814 depends on this PR. This PR does not fix jetbrains.jdk-17 and therefore src build fails. IMO, this PR should just cherry-pick the commit in #425814 and merge them together if we decide to pick this solution. 

#425529 is a dup. TBH I don't know which one is better. I'm fine with merging either one.

theCapypara commented:
As I understood it the solution in this PR has some issue with JCEF, but the other PR (https://github.com/NixOS/nixpkgs/pull/425529) has other issues. I see there is discussion going on and comparing it with the solution in this PR already.

@ners Could you also have a look at that PR and compare it with your solution and maybe give your opinions as well?

In any case I think all fixes related to jetbrains-jdk should go there first and that be merged first, then we continue here IMO.

I'm unfortunately not at all experienced with Java or really understand the changes that broke the jetbrains-jdk, so I can't really give a good opinion here myself on what is the best way to deal with these issues.

As for the discontinued IDEs, I've made a new issue as a reminder (https://github.com/NixOS/nixpkgs/issues/426427), if somebody has time feel free to open a PR, otherwise I'll do it soon :tm:

lylythechosenone commented:
No idea if this helps, but I've been using this for a week or so now and it seems perfectly stable (aarch64-darwin).

jamesward commented:
Just a heads up that new versions of IDEs are out (Ultimate 2025.1.4.1) so it'd be great to re-run the update script before this merges.

ners committed:
jetbrains.*: add missing build inputs

theCapypara reviewed (changes requested):
I'm getting the same build failures unfortunately.

ners committed:
jetbrains: 2025.1.1 -> 2025.1.5

jetbrains.clion: 2025.1.2 -> 2025.1.4
jetbrains.dataspell: 2025.1.1 -> 2025.1.2.1
jetbrains.goland: 2025.1.2 -> 2025.1.4
jetbrains.idea-community: 2025.1.2 -> 2025.1.4.1
jetbrains.idea-ultimate: 2025.1.2 -> 2025.1.4.1
jetbrains.phpstorm: 2025.1.2 -> 2025.1.4.1
jetbrains.pycharm-community: 2025.1.2 -> 2025.1.3.1
jetbrains.pycharm-professional: 2025.1.2 -> 2025.1.3.1
jetbrains.rider: 2025.1.2 -> 2025.1.4
jetbrains.ruby-mine: 2025.1.2 -> 2025.1.4.1
jetbrains.rust-rover: 2025.1.3 -> 2025.1.5
jetbrains.webstorm: 2025.1.2 -> 2025.1.4.1

ners committed:
jetbrains.plugins: update

jamesward commented:
Tested `idea-ultimate` on nixos and it worked.

JBassett commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 419026`
Commit: `a72819b139388a16b85dcc7b8117c154aec4016b`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 19 packages built:</summary>
  <ul>
    <li>jetbrains.aqua</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
  </ul>
</details>

JBassett commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 419026`
Commit: `a72819b139388a16b85dcc7b8117c154aec4016b`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 19 packages built:</summary>
  <ul>
    <li>jetbrains.aqua</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
  </ul>
</details>

JBassett commented:
Also tested `idea-ultimate` and it's working well.

theCapypara commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 419026`
Commit: `a72819b139388a16b85dcc7b8117c154aec4016b`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 19 packages built:</summary>
  <ul>
    <li>jetbrains.aqua</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
  </ul>
</details>

theCapypara reviewed (approved):
Thanks ners! In general this good to merge for me now, however I'm not sure if we have a plan on how to proceed with the -src build errors on aarch64. https://github.com/NixOS/nixpkgs/pull/425814 is closed now, is there a follow-up PR for that? I'm guessing the -src builds are broken on unstable too and it's unrelated to this PR - does somebody know or should we re-check that before merging?

aoli-al commented:
The aarch64 build failure should be addressed by #430439
